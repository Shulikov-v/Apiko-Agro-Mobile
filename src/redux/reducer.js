import { combineReducers } from 'redux';

import OrganizationReducer from '../scenes/organization/OrganizationState';
import LocalitiesReducer from '../scenes/localities/LocalitiesState';
import ProfileReducer from '../scenes/profile/ProfileState';

import NavigatorStateReducer from '../modules/navigator/NavigatorState';
import SessionStateReducer, { RESET_STATE } from '../modules/session/SessionState';
import AuthorizationReducer from '../scenes/authorization/AuthorizationState';

const appReducer = combineReducers({
  organization: OrganizationReducer,
  localities: LocalitiesReducer,
  profile: ProfileReducer,

  auth: AuthorizationReducer,
  session: SessionStateReducer,
  navigatorState: NavigatorStateReducer
});

export default (state, action) => {
  // if (action.type === RESET_STATE) {
  //   state = null; // eslint-disable-line no-param-reassign
  // }

  return appReducer(state, action);
};
