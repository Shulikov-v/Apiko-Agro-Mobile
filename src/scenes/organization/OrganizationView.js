import React, { PropTypes } from 'react';
import { View, StyleSheet } from 'react-native';
import { List, ListItem, Text, Card } from 'react-native-elements';

const OrganizationView = ({ organization }) => {
  const getLocalityListItem = (l, i) => (
    <ListItem
      key={i}
      title={l}
      hideChevron
    />
  );

  return (
    <View>
      <View style={styles.container}>
        <Text h3>{organization.name}</Text>
      </View>

      <Card title="Населені пункти" containerStyle={styles.cardContainer}>
        <List containerStyle={styles.list}>
          { organization.localities.map(getLocalityListItem) }
        </List>
      </Card>
    </View>
  );
};

OrganizationView.propTypes = {
  organization: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  container: { marginTop: 10, padding: 15 },
  cardContainer: { borderRadius: 3 },
  list: { marginTop: 0 },
});

export default OrganizationView;
