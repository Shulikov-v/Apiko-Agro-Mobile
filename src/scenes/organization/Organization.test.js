import React from 'react';
import renderer from 'react-test-renderer';

import reducer, { getOrganizationInfo } from './OrganizationState';
import OrganizationView from './OrganizationView';

describe('Organization', function() {
  const organizationMock = {
    _id: '123',
    name: 'Organization',
    localities: ['loc1', 'loc2']
  };

  describe('Thunks', function() {

    describe('getOrganizatonInfo', function() {

      it('should return a function', function() {
        expect(getOrganizationInfo()).toBeInstanceOf(Function);
      })

    })

  });

  describe('Reducer', function() {

    it('should return initialized organization', function() {
      const actionMock = { type: 'INIT_ORGANIZATION', organization: organizationMock };

      expect(reducer(null, actionMock)).toBe(organizationMock);
    })

  });

  describe('View', function() {

    it('renders correctly', function() {

      const tree = renderer.create(
        <OrganizationView organization={organizationMock}/>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

  });

});
