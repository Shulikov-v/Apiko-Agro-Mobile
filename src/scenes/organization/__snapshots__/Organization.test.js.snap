// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization View renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "marginTop": 10,
        "padding": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          Object {
            "fontSize": 32.199999999999996,
          },
          undefined,
          undefined,
          undefined,
          Object {},
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Organization
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 15,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
          undefined,
        ]
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontSize": 16.099999999999998,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Населені пункти
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#e1e8ee",
                "height": 1,
              },
              Array [
                Object {
                  "marginBottom": 15,
                },
                undefined,
              ],
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 1,
              "borderColor": "#cbd2d9",
              "borderTopWidth": 1,
              "marginTop": 20,
            },
            Object {
              "marginTop": 0,
            },
          ]
        }
      >
        <View
          onLongPress={undefined}
          onPress={undefined}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 16.099999999999998,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  loc1
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
          </View>
        </View>
        <View
          onLongPress={undefined}
          onPress={undefined}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 16.099999999999998,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  loc2
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
